src/cephadm/cephadm.py

```diff
@@ -2998,6 +2998,18 @@ def fetch_meta(ctx: CephadmContext) -> Dict[str, Any]:
     return {}
 
 
+def fetch_configs(ctx: CephadmContext) -> Dict[str, str]:
+    cfg_blobs = getattr(ctx, 'config_blobs', None)
+    if cfg_blobs:
+        cfg_blobs = dict(cfg_blobs)
+        cfg_blobs.pop('custom_config_files', None)
+        return cfg_blobs
+    cfg_json = getattr(ctx, 'config_json', None)
+    if cfg_json:
+        return get_parm(cfg_json) or {}
+    return {}
```

Comment:
It is set w/in cephadm.py - [in what eventually becomes apply_deploy_config_to_ctx](https://github.com/ceph/ceph/pull/51738/files#diff-2bc083a616d69767b3cb29ad7910698521b38a2c7d078bc7f1401b76d07ef9faR6238). The idea here is that if `config_blobs` is set then you have already turned JSON into python objects (dicts, etc). If you only have `config_json` then it's likely you have a JSON-string that must be processed using `get_parm` [1]. I had thought about assigning the results of `get_parm` back to `cfg_blobs` similar to the approach in `fetch_meta` but since `get_parm` is a bit complex I decided not to do that in order to keep things simpler.

As a happy accident looking for this made me find another mistake (the line below the linked line). :-)

[1] I constantly mistake this as `get_param` and confuse myself.